packages/ember-glimmer/tests/integration/components/curly-components-test.js

```diff
@@ -917,4 +921,977 @@ moduleFor('Components test: curly components', class extends RenderingTest {
+      template: 'somecomponent'
+    });
+
+    this.render(
+      '{{somecomponent}}',
+      {
+        'somecomponent': 'notsomecomponent'
+      }
+    );
+
+    this.assertText('notsomecomponent');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('notsomecomponent');
+
+    this.runTask(() => this.context.set('somecomponent', 'not not notsomecomponent'));
+
+    this.assertText('not not notsomecomponent');
+
+    this.runTask(() => this.context.set('somecomponent', 'notsomecomponent'));
+
+    this.assertText('notsomecomponent');
+  }
+
+  ['@htmlbars GlimmerComponent cannot be invoked with curly braces']() {
+    this.registerComponent('non-block', {
+      ComponentClass: GlimmerComponent.extend(),
+      template: 'In layout'
+    });
+
+    expectAssertion(() => {
+      this.render('{{non-block}}');
+    }, /cannot invoke the 'non-block' component with curly braces/);
+  }
+
+  ['@test non-block with properties on attrs']() {
+    this.registerComponent('non-block', {
+      template: 'In layout - someProp: {{attrs.someProp}}'
+    });
+
+    this.render('{{non-block someProp="something here"}}');
+
+    this.assertText('In layout - someProp: something here');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('In layout - someProp: something here');
+  }
+
+  ['@test non-block with properties on attrs and component class']() {
+    this.registerComponent('non-block', {
+      ComponentClass: Component.extend(),
+      template: 'In layout - someProp: {{attrs.someProp}}'
+    });
+
+    this.render('{{non-block someProp="something here"}}');
+
+    this.assertText('In layout - someProp: something here');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('In layout - someProp: something here');
+  }
+
+  ['@htmlbars rerendering component with attrs from parent'](assert) {
+    let willUpdate = 0;
+    let didReceiveAttrs = 0;
+
+    this.registerComponent('non-block', {
+      ComponentClass: Component.extend({
+        didReceiveAttrs() {
+          didReceiveAttrs++;
+        },
+
+        willUpdate() {
+          willUpdate++;
+        }
+      }),
+      template: 'In layout - someProp: {{someProp}}'
+    });
+
+    this.render('{{non-block someProp=someProp}}', {
+      someProp: 'wycats'
+    });
+
+    assert.equal(didReceiveAttrs, 1, 'The didReceiveAttrs hook fired');
+    this.assertText('In layout - someProp: wycats');
+
+    this.runTask(() => this.context.set('someProp', 'tomdale'));
+
+    this.assertText('In layout - someProp: tomdale');
+    assert.equal(didReceiveAttrs, 2, 'The didReceiveAttrs hook fired again');
+    assert.equal(willUpdate, 1, 'The willUpdate hook fired once');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('In layout - someProp: tomdale');
+    assert.equal(didReceiveAttrs, 3, 'The didReceiveAttrs hook fired again');
+    assert.equal(willUpdate, 2, 'The willUpdate hook fired again');
```

Comment:
Missing the "R" (reset) step – set `someProp` back to `wycats` and add the relevant assertions.